storage/local/chunk/chunk.go

```diff
@@ -57,96 +66,96 @@ func (ce *ChunkEncoding) Set(s string) error {
-// eviction is protected by a mutex. This affects the following methods: pin,
-// unpin, refCount, isEvicted, maybeEvict. These methods can be called at any
+// eviction is protected by a mutex. This affects the following methods: Pin,
+// Unpin, RefCount, IsEvicted, MaybeEvict. These methods can be called at any
 // time without further prerequisites.
 //
 // Another group of methods acts on (or sets) the underlying chunk. These
 // methods involve no locking. They may only be called if the caller has pinned
 // the chunk (to guarantee the chunk is not evicted concurrently). Also, the
 // caller must make sure nobody else will call these methods concurrently,
-// either by holding the sole reference to the chunkDesc (usually during loading
-// or creation) or by locking the fingerprint of the series the chunkDesc
-// belongs to. The affected methods are: add, maybePopulateLastTime, setChunk.
+// either by holding the sole reference to the ChunkDesc (usually during loading
+// or creation) or by locking the fingerprint of the series the ChunkDesc
+// belongs to. The affected methods are: Add, MaybePopulateLastTime, SetChunk.
 //
-// Finally, there are the special cases firstTime and lastTime. lastTime requires
+// Finally, there are the special cases FirstTime and LastTime. LastTime requires
 // to have locked the fingerprint of the series but the chunk does not need to
-// be pinned. That's because the chunkLastTime field in chunkDesc gets populated
+// be pinned. That's because the ChunkLastTime field in ChunkDesc gets populated
 // upon completion of the chunk (when it is still pinned, and which happens
 // while the series's fingerprint is locked). Once that has happened, calling
-// lastTime does not require the chunk to be loaded anymore. Before that has
-// happened, the chunk is pinned anyway. The chunkFirstTime field in chunkDesc
-// is populated upon creation of a chunkDesc, so it is alway safe to call
-// firstTime. The firstTime method is arguably not needed and only there for
-// consistency with lastTime.
-type ChunkDesc struct {
+// LastTime does not require the chunk to be loaded anymore. Before that has
+// happened, the chunk is pinned anyway. The ChunkFirstTime field in ChunkDesc
+// is populated upon creation of a ChunkDesc, so it is alway safe to call
+// FirstTime. The FirstTime method is arguably not needed and only there for
+// consistency with LastTime.
+type Desc struct {
 	sync.Mutex           // Protects pinning.
-	c              Chunk // nil if chunk is evicted.
+	C              Chunk // nil if chunk is evicted.
 	rCnt           int
-	chunkFirstTime model.Time // Populated at creation. Immutable.
-	chunkLastTime  model.Time // Populated on closing of the chunk, model.Earliest if unset.
+	ChunkFirstTime model.Time // Populated at creation. Immutable.
+	ChunkLastTime  model.Time // Populated on closing of the chunk, model.Earliest if unset.
```

Comment:
The main reason for not simply calling these `firstTime` and `lastTime` was that those names would have collided with the respective method names. Now that the method names are exported, I'd say you should totally name the fields `firstTime` and `lastTime`.